jobs/BnRStatusPollerJob.js

```diff
@@ -92,26 +92,55 @@ class BnRStatusPollerJob extends BaseJob {
               }
             });
           })
-          .then(() => eventmesh.apiServerClient.getResource({
-            resourceGroup: CONST.APISERVER.RESOURCE_GROUPS.BACKUP,
-            resourceType: CONST.APISERVER.RESOURCE_TYPES.DEFAULT_BACKUP,
-            resourceId: backup_guid
+          .then(() => lockManager.lock(instance_guid, {
+            lockedResourceDetails: {
+              resourceGroup: CONST.APISERVER.RESOURCE_GROUPS.BACKUP,
+              resourceType: CONST.APISERVER.RESOURCE_TYPES.DEFAULT_BACKUP,
+              resourceId: backup_guid,
+              operation: CONST.OPERATION_TYPE.BACKUP
+            }
           }))
-          .catch(NotFound, () => {
-            logger.info(`Creating missing operation resource for backup ${backup_guid}`);
-            return eventmesh.apiServerClient.createResource({
+          .catch(err => {
+            if (err instanceof DeploymentAlreadyLocked) {
```

Comment:
use the error type in catch itself instead of such checks like above. Also this might seem unnecessary as per above comment.